File: backend/NemesiLIB/Model/Utente.cs
using Microsoft.AspNetCore.Identity;

namespace NemesiLIB.Model
{
    public class Utente : IdentityUser
    {
        public string Nominativo { get; set; }
        public string? Societa { get; set; }
        public bool IsEsterno { get; set; }
        public decimal CostoOrario { get; set; }
        public decimal CostoKmAuto { get; set; }

        public string? RuoloAziendale { get; set; }

        public Utente(string email, string nominativo, bool isEsterno = false, string societa = null, decimal costoOrario = 0m, decimal costoKmAuto = 0m, string ruoloAziendale = null)
        {
            Email = email;
            UserName = email;
            Nominativo = nominativo;
            IsEsterno = isEsterno;
            Societa = societa;
            CostoOrario = costoOrario;
            CostoKmAuto = costoKmAuto;
            RuoloAziendale = ruoloAziendale;
        }
    }
}

File: frontend/gestionale-bertozzi/src/app/models/utente.ts
import { Ruolo } from "./ruolo";

export class Utente {
    id?: string;
    nominativo?: string;
    userName?: string;
    email?: string;
    ruoli?: string[];
    isEsterno: boolean = false;
    isAttivo: boolean = true;
    societa?: string;
    costoOrario?: number;
    costoKmAuto?: number;
    ruoloAziendale?: string;

    static map(utente: any): Utente {
        let u = Object.assign(new Utente(), utente) as Utente;
        return u;
    }

    static mapArray(arr: any[]): Utente[] {
        return (arr || []).map(u => Utente.map(u));
    }
}
